File: dev_server.py
import http.server
import socketserver
import os

class CatchAllHttpRequestHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        # Redirect all requests to the index.html file
        self.path = 'index.html'
        return http.server.SimpleHTTPRequestHandler.do_GET(self)

# Set the port
PORT = 8000

handler_object = CatchAllHttpRequestHandler

my_server = socketserver.TCPServer(("", PORT), handler_object)

# Start the server
print(f"Serving at port {PORT}")
my_server.serve_forever()

File: devServer.js
const http = require("http");
const fs = require("fs");
const path = require("path");

// Path to your index.html file
const indexPath = path.join(__dirname, "index.html");

const server = http.createServer((req, res) => {
  fs.readFile(indexPath, (error, content) => {
    if (error) {
      // If there's an error reading the file, log it and return a simple error message
      console.error("Error serving the index.html file:", error);
      res.writeHead(500, { "Content-Type": "text/plain" });
      res.end("Internal server error");
    } else {
      // Successfully read the file, serve it as HTML
      res.writeHead(200, { "Content-Type": "text/html" });
      res.end(content, "utf-8");
    }
  });
});

const port = 8000; // Or any port you'd like to use
server.listen(port, () => {
  console.log(`Development server running at http://localhost:${port}/`);
});
